eslint.config.js

```diff
@@ -37,6 +41,63 @@ export default [
           varsIgnorePattern: '^_',
           caughtErrorsIgnorePattern: '^_'
         }
+      ],
+      'import/export': 'error', // forbid invalid exports and re-exports of the same name
+      'import/no-duplicates': 'error', // forbid duplicate imports
+      'import/order': [
+        'error',
+        {
+          'groups': [
+            'builtin', // node built-in modules
+            'external', // npm packages
+            'internal', // @shared/**, @tools/**, @/** and ~/** aliases
+            ['parent', 'sibling', 'index'], // relative imports
+            'type'
+          ],
+          'pathGroups': [
+            {
+              pattern: 'react',
+              group: 'external',
+              position: 'before'
+            },
+            {
+              pattern: 'react-router',
+              group: 'external',
+              position: 'before'
+            },
+            {
+              pattern: '@**',
+              group: 'external',
+              position: 'after'
+            },
+            {
+              pattern: '@shared/**',
+              group: 'internal',
+              position: 'before'
+            },
+            {
+              pattern: '@tools/**',
+              group: 'internal',
+              position: 'before'
+            },
+            {
+              pattern: '@/**',
+              group: 'internal',
+              position: 'after'
+            },
+            {
+              pattern: '~/**',
+              group: 'internal',
+              position: 'after'
+            }
+          ],
+          'pathGroupsExcludedImportTypes': ['builtin'],
```

Comment:
```suggestion
          'pathGroupsExcludedImportTypes': ['builtin', 'type'],
```

Not sure if we can have the best of both worlds? Adding to the array moves all type imports to bottom of file, e.g. [OptionSelector.tsx](frontend/app/components/redesign/components/OptionSelector.tsx), but would break some of the import ordering in that file for example it moves a type import to bottom, but it's from `'react'` which we want at top

Reply:
I chose to add `type` to `pathGroupsExcludedImportTypes` and have all type imports at the end, without breaking the order of the other imports.

The trade-off is that we cannot apply the same order of path groups to the `type` subgroup, only alphabetical order